packages/app-mobile/utils/fs-driver/fs-driver-rn.ts

```diff
@@ -285,6 +286,10 @@ export default class FsDriverRN extends FsDriverBase {
 	}
 
 	public async readFileChunk(handle: any, length: number, rawEncoding = 'base64') {
+		if (!handle?.stat) {
+			throw new JoplinError('File does not exist (reading file chunk).', 'ENOENT');
+		}
```

Comment:
I'm throwing the exception here, rather than in `open` to avoid [parsing the file system flags](https://nodejs.org/api/fs.html#file-system-flags) argument to `open`.

Reply:
Thanks Henry, the fix makes sense. What message will the synchronizer print in case of error now? Is it possible to improve it too with something like "Trying to encrypt resource, but only metadata is present" with the resource title or ID? (only assuming it's a simple change - if there's no access to title or ID at this point, we can leave it as is)

Reply:
Initially, it doesn't show an error message, just a "Some items cannot be synchronized banner":
![image](https://github.com/laurent22/joplin/assets/46334387/24d4034a-16e3-4da0-9f50-dcbb18327f7e)

# Before changing the message

After clicking on the banner,
![image](https://github.com/laurent22/joplin/assets/46334387/d175e7c8-4b43-4948-b952-bc9c958dbb3f)

# After changing the message

**Mobile**:

![image](https://github.com/laurent22/joplin/assets/46334387/ba9ced97-b794-4837-a734-a26326dccc72)

**Desktop**:
![image](https://github.com/laurent22/joplin/assets/46334387/9bdc66e9-7dc3-45a1-8251-ec6a236badcd)